spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/IntegrationFlowFunctionSupport.java

```diff
@@ -0,0 +1,138 @@
+	 * @param functionInspector
+	 * @param messageConverterFactory
+	 * @param functionProperties
+	 */
+	public IntegrationFlowFunctionSupport(FunctionCatalog functionCatalog, FunctionInspector functionInspector,
+			CompositeMessageConverterFactory messageConverterFactory, FunctionProperties functionProperties) {
+
+		Assert.notNull(functionCatalog, "'functionCatalog' must not be null");
+		Assert.notNull(functionInspector, "'functionInspector' must not be null");
+		Assert.notNull(messageConverterFactory, "'messageConverterFactory' must not be null");
+		this.functionCatalog = functionCatalog;
+		this.functionInspector = functionInspector;
+		this.messageConverterFactory = messageConverterFactory;
+		this.functionProperties = functionProperties;
+	}
+
+	/**
+	 * Creates an instance of the {@link IntegrationFlowBuilder} from a {@link Supplier} bean available in the context.
+	 * The name of the bean must be provided via `spring.cloud.stream.function.name` property.
+	 *
+	 * @return instance of {@link IntegrationFlowBuilder}
+	 */
+	public <O> IntegrationFlowBuilder integrationFlowFromNamedSupplier() {
+		if (StringUtils.hasText(this.functionProperties.getName())) {
+			Supplier<?> supplier = functionCatalog.lookup(Supplier.class, this.functionProperties.getName());
+			if (supplier instanceof FluxSupplier) {
+				supplier = ((FluxSupplier<?>)supplier).getTarget();
+			}
+			return this.integrationFlowFromProvidedSupplier(supplier);
+		}
+		throw new IllegalStateException("Supplier name is not provided");
+	}
+
+	/**
+	 * Creates an instance of the {@link IntegrationFlowBuilder} from a provided {@link Supplier}.
+	 *
+	 * @return instance of {@link IntegrationFlowBuilder}
+	 */
+	public <O> IntegrationFlowBuilder integrationFlowFromProvidedSupplier(Supplier<?> supplier) {
+		IntegrationFlowBuilder flowBuilder = IntegrationFlows.from(supplier);
+		return flowBuilder;
+	}
+
+	/**
+	 * Adds a {@link Function} bean to the end of an integration flow.
+	 * The name of the bean must be provided via `spring.cloud.stream.function.name` property.
+	 * <p>
+	 * NOTE: If this method returns true, the integration flow is now represented as a Reactive Streams {@link Publisher} bean.
+	 * </p>
+	 * @param flowBuilder instance of the {@link IntegrationFlowBuilder} representing the current state of the integration flow
+	 * @param outputChannel channel where the output of a function will be sent
+	 * @return true if {@link Function} was located and added and false if it wasn't.
+	 */
+	public <I,O> boolean andThenFunction(IntegrationFlowBuilder flowBuilder, MessageChannel outputChannel) {
+		if (StringUtils.hasText(this.functionProperties.getName())) {
+			FunctionInvoker<I,O> functionInvoker =
+					new FunctionInvoker<>(this.functionProperties.getName(), this.functionCatalog, this.functionInspector,
+							this.messageConverterFactory);
+
+			subscribeToInput(functionInvoker, flowBuilder.toReactivePublisher(), message -> outputChannel.send(message));
```

Comment:
Nit-pick: the method reference can be used instead: `outputChannel::send`.
At least this is what my IDEA suggests me. No?

Reply:
yes